functional_tests.go

```diff
@@ -121,6 +121,33 @@ func ignoredLog(function string, args map[string]interface{}, startTime time.Tim
 	return log.WithFields(fields)
 }
 
+// Delete objects in given bucket, recursively
+func cleanupBucket(bucketName string, c *minio.Client) error {
+	// Create a done channel to control 'ListObjectsV2' go routine.
+	doneCh := make(chan struct{})
+	// Exit cleanly upon return.
+	defer close(doneCh)
+	// Iterate over all objects in the bucket via listObjectsV2 and delete
+	for objCh := range c.ListObjectsV2(bucketName, "", true, doneCh) {
+		if objCh.Key != "" {
+			err := c.RemoveObject(bucketName, objCh.Key)
+			if err != nil {
+				return err
+			}
+		} else {
+			if objCh.Err != nil {
+				return objCh.Err
+			}
+		}
+	}
+	// objects are already deleted, clear the buckets now
+	err := c.RemoveBucket(bucketName)
+	if err != nil {
+		return err
+	}
+	return err
+}
```

Comment:
what if there are incomplete upload parts? that would need cleanup as well.

Reply:
can you explain a bit more @poornas

Reply:
if there are incomplete uploads to a bucket, RemoveBucket would fail. For tests using large uploads, shouldn't the cleanup also remove incomplete uploads if any?

Reply:
done @poornas

Reply:
RemoveBucket works fine on all gateways even with incomplete uploads, that is the S3 behavior. But it is a good change since it will cater for B2 gateway backend as well. Which requires explicit removal of incomplete uploads.